e2e/testenv/infra/go.mod

```diff
@@ -2,7 +2,7 @@ module github.com/rancher/fleet/e2e/testenv/infra
 
 go 1.21
 
-toolchain go1.21.0
+toolchain go1.22.7
```

Comment:
Should we consider bumping Go in other `go.mod` files, from 1.21 and 1.19 to 1.22.7?